pkg/restore/change_image_name_action.go

```diff
@@ -0,0 +1,260 @@
+func NewChangeImageNameAction(
+	logger logrus.FieldLogger,
+	configMapClient corev1client.ConfigMapInterface,
+) *ChangeImageNameAction {
+	return &ChangeImageNameAction{
+		logger:          logger,
+		configMapClient: configMapClient,
+	}
+}
+
+// AppliesTo returns the resources that ChangeImageNameAction should
+// be run for.
+func (a *ChangeImageNameAction) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"deployments", "statefulsets", "daemonsets", "replicasets", "replicationcontrollers", "jobs", "cronjobs", "pods"},
+	}, nil
+}
+
+// Execute updates the item's spec.containers' image if a mapping is found
+// in the config map for the plugin.
+func (a *ChangeImageNameAction) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
+	a.logger.Info("Executing ChangeImageNameAction")
+	defer a.logger.Info("Done executing ChangeImageNameAction")
+
+	config, err := getPluginConfig(common.PluginKindRestoreItemAction, "velero.io/change-image-repository", a.configMapClient)
+	if err != nil {
+		return nil, err
+	}
+
+	if config == nil || len(config.Data) == 0 {
+		a.logger.Info("No image name mappings found")
+		return velero.NewRestoreItemActionExecuteOutput(input.Item), nil
+	}
+
+	obj, ok := input.Item.(*unstructured.Unstructured)
+	if !ok {
+		return nil, errors.Errorf("object was of unexpected type %T", input.Item)
+	}
+
+	log := a.logger.WithFields(map[string]interface{}{
+		"kind":      obj.GetKind(),
+		"namespace": obj.GetNamespace(),
+		"name":      obj.GetName(),
+	})
+
+	if obj.GetKind() == "Pod" {
+		pod := new(corev1.Pod)
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), pod); err != nil {
+			a.logger.Infof("UnstructuredConverter meet error: %v", err)
+			return nil, err
+		}
+		needUpdateObj := false
+
+		if len(pod.Spec.Containers) > 0 {
+			for i, container := range pod.Spec.Containers {
+				if exists, newImageName, err := a.isImageReplaceRuleExist(log, container.Image, config); exists && err == nil {
+					needUpdateObj = true
+					a.logger.Infof("Updating item's image from %s to %s", container.Image, newImageName)
+					pod.Spec.Containers[i].Image = newImageName
+				}
+			}
+		}
+
+		if len(pod.Spec.InitContainers) > 0 {
+			for i, container := range pod.Spec.InitContainers {
+				if exists, newImageName, err := a.isImageReplaceRuleExist(log, container.Image, config); exists && err == nil {
+					needUpdateObj = true
+					a.logger.Infof("Updating item's image from %s to %s", container.Image, newImageName)
+					pod.Spec.InitContainers[i].Image = newImageName
+				}
+			}
+		}
+		if needUpdateObj {
+			newObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(pod)
+			if err != nil {
+				return nil, errors.Wrap(err, "convert obj to Pod failed")
+			}
+			obj.Object = newObj
+		}
+
+	} else if obj.GetKind() == "Job" || obj.GetKind() == "CronJob" {
```

Comment:
This section is for CronJob only -- a CronJob has a field Spec.JobTemplate which contains a JobSpec (so spec/jobTemplate/spec/template/spec... as below). A Job just has the PodTemplateSpec directly at Spec.Template, which is the same as Deployment, etc. "Job" should be removed from this conditional so the Job case will fall through to the final else clause.

Reply:
@sseago 
thanks for your careful review.
Modification completed.